### PR TITLE
xplat: fix the path for i18n

### PR DESCRIPTION
**Pull request by obastemur**

### CODE REVIEW

**Comment by obastemur** [comment]
BTW; I'm still waiting the build on my end.

@dilijev can you please try with node-chakracore as well ?

**Comment by obastemur** [comment]
@dotnet-bot test Ubuntu static_ubuntu_linux_test please

**Comment by obastemur** [comment]
FYI; my end build with `--with-intl` looks fine.

File: CMakeLists.txt [hunk 1]
```diff
@@ -126,11 +126,24 @@ function(clr_unknown_arch)
...
+
 if(ICU_INCLUDE_PATH)
     add_definitions(-DHAS_REAL_ICU=1)
     set(ICU_CC_PATH "${ICU_INCLUDE_PATH}/../lib/")
     find_library(ICUUC icuuc PATHS ${ICU_CC_PATH} NO_DEFAULT_PATH)
-    find_library(ICU18 icui18n PATHS ${ICU_CC_PATH} NO_DEFAULT_PATH)
+    if(ICU_INTL_ENABLED)
+        find_library(ICU18 icui18n PATHS ${ICU_CC_PATH} NO_DEFAULT_PATH)
+        # icu header files are either located under the same folder or i18n is under a relative path
+        # TODO (unlikely): shall we add `--icu-intl` to build.sh ?
```
Comment (dilijev) [inline CMakeLists.txt]: Are we making a distinction between --icu-intl (as in the path we want to specify) versus --intl-icu (not present on CMake builds, terminology used on Windows builds i.e. /p:IntlICU=true), versus --with-intl?

I agree this is not likely necessary and can be skipped for now.

File: CMakeLists.txt [hunk 2]
```diff
@@ -126,11 +126,24 @@ function(clr_unknown_arch)
...
+        set(ICU_INCLUDE_PATH
+          "${ICU_INCLUDE_PATH}"
+          "${ICU_INCLUDE_PATH}/../i18n/")
```
Comment (dilijev) [inline CMakeLists.txt]: This seems reasonable to me. @jackhorton look good to you?

**Comment by dilijev** [comment]
@jackhorton You're more familiar with the Node-ChakraCore workflow, so can you give this a try there and let me know? I'll give it a try as well.

**Comment by dilijev** [comment]
@obastemur can you please retarget this PR to `intl-icu` (which will be merged to release/1.7 once all the other PRs to `intl-icu` land?

**Comment by dilijev** [comment]
@obastemur nevermind, I forgot I could do that for you, and since intl-icu is ahead of 1.7 at the moment, there is no strangeness.

**Pull request merged**